frame/contracts-registry/src/lib.rs

```diff
@@ -0,0 +1,273 @@
+#![recursion_limit = "256"]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+pub mod benchmarking;
+pub mod mock;
+pub mod tests;
+pub mod types;
+pub mod weights;
+
+pub use types::{ContractDataStorage, ContractIndex};
+pub use weights::WeightInfo;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::OriginFor;
+
+	/// The current storage version.
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+	#[pallet::pallet]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + Sized {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// Maximum length of contract data.
+		#[pallet::constant]
+		type MaxContractContentLen: Get<u32>;
+
+		/// Maximum signatures per contract
+		#[pallet::constant]
+		type MaxSignatures: Get<u32>;
+
+		/// Maximum judges count
+		#[pallet::constant]
+		type MaxJudges: Get<u32>;
+
+		/// Origin from which a judge may be added
+		type AddJudgeOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// Origin from which the new typical action can be made.
+		///
+		/// The success variant is the account id of the depositor.
```

Comment:
I don't think there's a concept of `depositor` in this pallet. Should it be `submitter` or smth?